app/src/main/java/com/github/kulebin/myfoursquareapp/dataSource/DataLoader.java

```diff
@@ -0,0 +1,58 @@
+package com.github.kulebin.myfoursquareapp.dataSource;
+
+import android.os.Handler;
+
+import com.github.kulebin.myfoursquareapp.http.IHttpClient;
+
+import java.io.IOException;
+
+class DataLoader implements IDataLoader {
+
+    //todo should be initialized or pass as param
+    private IDataCache mDataCache;
+    private IProcessor mProcessor;
+
+    @Override
+    public <Result> void loadData(final String url, final IOnResultCallback<Result> pOnResultCallback) {
+
+        if (mDataCache.isDataCached(url)) {
```

Comment:
Let keep this part for future, when caching will be introduced.
Now code should work if no mDataCache was created